sources/packages/backend/apps/api/src/route-controllers/application-restriction-bypass/_tests_/e2e/application-restriction-bypass.aest.controller.getApplicationRestrictionBypass.e2e-spec.ts

```diff
@@ -0,0 +1,130 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import {
+  E2EDataSources,
+  RestrictionCode,
+  createE2EDataSources,
+  createFakeUser,
+  saveFakeApplication,
+  saveFakeApplicationRestrictionBypass,
+} from "@sims/test-utils";
+import {
+  AESTGroups,
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAESTToken,
+} from "../../../../testHelpers";
+import * as request from "supertest";
+import { OfferingIntensity, RestrictionActionType, User } from "@sims/sims-db";
+import { getUserFullName } from "../../../../utilities";
+
+describe("ApplicationRestrictionBypassAESTController(e2e)-getApplicationRestrictionBypass.", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+  let sharedMinistryUser: User;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    sharedMinistryUser = await db.user.save(createFakeUser());
+  });
+
+  it("Should get an application restriction bypass for a submitted part-time application when there is an application restriction bypass with the required id.", async () => {
+    // Arrange
+    const application = await saveFakeApplication(db.dataSource, undefined, {
+      offeringIntensity: OfferingIntensity.partTime,
+    });
```

Comment:
Suggestion, a new application in being created many times to be provided to `saveFakeApplicationRestrictionBypass` and it can be avoided if the `saveFakeApplicationRestrictionBypass` creates as an application if one is not provided.